src/admin/apiProvider/dataProviders/projectReportDataProvider.ts

```diff
@@ -1,48 +1,30 @@
-import { DataProvider } from "react-admin";
+import { DataProvider, HttpError } from "react-admin";
 
-import {
-  DeleteV2AdminProjectReportsUUIDError,
-  fetchDeleteV2AdminProjectReportsUUID,
-  fetchGetV2AdminProjectReports,
-  fetchGetV2ENTITYUUID,
-  GetV2AdminProjectReportsError,
-  GetV2ENTITYUUIDError
-} from "@/generated/apiComponents";
+import { loadFullProjectReport, loadProjectReportIndex } from "@/connections/Entity";
+import { DeleteV2AdminProjectReportsUUIDError, fetchDeleteV2AdminProjectReportsUUID } from "@/generated/apiComponents";
 
 import { getFormattedErrorForRA } from "../utils/error";
-import { apiListResponseToRAListResult, raListParamsToQueryParams } from "../utils/listing";
-
-const projectReportSortableList = ["title", "project_name", "organisation_name", "due_at", "submitted_at"];
+import { entitiesListResult, raConnectionProps } from "../utils/listing";
 
 // @ts-ignore
 export const projectReportDataProvider: DataProvider = {
+  // @ts-expect-error until we can get the whole DataProvider on ProjectReportLightDto
   async getList(_, params) {
-    try {
-      const response = await fetchGetV2AdminProjectReports({
-        queryParams: raListParamsToQueryParams(params, projectReportSortableList)
-      });
-
-      return apiListResponseToRAListResult(response);
-    } catch (err) {
-      throw getFormattedErrorForRA(err as GetV2AdminProjectReportsError);
+    const connection = await loadProjectReportIndex(raConnectionProps(params));
+    if (connection.fetchFailure != null) {
+      throw new HttpError(connection.fetchFailure.message, connection.fetchFailure.statusCode);
```

Comment:
I added a utility for this pattern recently - `v3ErrorForRa` - see the projectDataProvider for an example.